src/test/binaryutils_test.cpp

```diff
@@ -215,4 +215,94 @@ TEST_F(BinaryUtilsTest, positive_resizing) {
+    ASSERT_EQ(blob->size(), 1);
+    ASSERT_EQ(status, ovms::StatusCode::OK);
+    uint8_t* ptr = blob->buffer();
+    ASSERT_EQ(std::equal(ptr, ptr + blob->size(), grayscale_expected_blob), true);
+    tensorflow::TensorProto stringValDeserialized;
+    convertBlobToStringVal(stringValDeserialized, tensorInfo, blob);
+    EXPECT_EQ(stringVal.string_val(0).substr(stringVal.string_val(0).find("\xFF\xC0")), stringValDeserialized.string_val(0).substr(stringValDeserialized.string_val(0).find("\xFF\xC0")));
+}
+
+TEST_F(BinaryUtilsTest, positive_rgb_serialization_deserialization) {
+    uint8_t rgb_expected_blob[] = {0x24, 0x1b, 0xed};
+
+    std::ifstream DataFile;
+    DataFile.open("/ovms/src/test/binaryutils/rgb.jpg", std::ios::binary);
+    DataFile.seekg(0, std::ios::end);
+    size_t filesize = DataFile.tellg();
+    DataFile.seekg(0);
+    std::unique_ptr<char[]> image_bytes(new char[filesize]);
+    DataFile.read(image_bytes.get(), filesize);
+
+    tensorflow::TensorProto stringVal;
+    InferenceEngine::Blob::Ptr blob;
+    stringVal.set_dtype(tensorflow::DataType::DT_STRING);
+    stringVal.add_string_val(image_bytes.get(), filesize);
+
+    std::shared_ptr<TensorInfo> tensorInfo = std::make_shared<TensorInfo>("", InferenceEngine::Precision::U8, shape_t{1, 3, 1, 1}, InferenceEngine::Layout::NCHW);
+
+    auto status = convertStringValToBlob(stringVal, &blob, tensorInfo);
+
+    ASSERT_EQ(blob->size(), 3);
+    ASSERT_EQ(status, ovms::StatusCode::OK);
+    uint8_t* ptr = blob->buffer();
+    ASSERT_EQ(std::equal(ptr, ptr + blob->size(), rgb_expected_blob), true);
+    tensorflow::TensorProto stringValDeserialized;
+    convertBlobToStringVal(stringValDeserialized, tensorInfo, blob);
+
+    EXPECT_EQ(stringVal.string_val(0).substr(stringVal.string_val(0).find("\xFF\xC0")), stringValDeserialized.string_val(0).substr(stringValDeserialized.string_val(0).find("\xFF\xC0")));
+}
+
+TEST_F(BinaryUtilsTest, positive_batch_size_2_serialization_deserialization) {
+    uint8_t rgb_batchsize_2_blob[] = {0x24, 0x1b, 0xed, 0x24, 0x1b, 0xed};
+
+    std::ifstream DataFile;
+    DataFile.open("/ovms/src/test/binaryutils/rgb.jpg", std::ios::binary);
+    DataFile.seekg(0, std::ios::end);
+    size_t filesize = DataFile.tellg();
+    DataFile.seekg(0);
+    std::unique_ptr<char[]> image_bytes(new char[filesize]);
+    DataFile.read(image_bytes.get(), filesize);
+
+    tensorflow::TensorProto stringVal;
+    InferenceEngine::Blob::Ptr blob;
+    stringVal.set_dtype(tensorflow::DataType::DT_STRING);
+    stringVal.add_string_val(image_bytes.get(), filesize);
+    stringVal.add_string_val(image_bytes.get(), filesize);
+
+    std::shared_ptr<TensorInfo> tensorInfo = std::make_shared<TensorInfo>("", InferenceEngine::Precision::U8, shape_t{2, 3, 1, 1}, InferenceEngine::Layout::NCHW);
+
+    auto status = convertStringValToBlob(stringVal, &blob, tensorInfo);
+
+    ASSERT_EQ(blob->size(), 6);
+    ASSERT_EQ(status, ovms::StatusCode::OK);
```

Comment:
This is wrong to test blob size before checking status. In case status is not OK we end up with `status` variable with nullptr value. Dereferencing nullptr causes segfault. We do not want segfaults in our unit tests in case of fails but meaningful messages wherever possible.